File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
def solution(queue1, queue2):
    q1 = sum(queue1)
    q2 = sum(queue2)
    
    total = q1 + q2
    if total % 2 != 0:
        return -1
    
    target = total // 2
    
    q = queue1 + queue2
    start = 0
    end = len(queue1) - 1
    answer = 0
    
    while q1 != target and start <= end:
        if q1 < target:
            end += 1
            if end == len(q)-1:
                return -1
            q1 += q[end]
        else:
            q1 -= q[start]
            start += 1
        answer += 1
    return answer 
        
            

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        long t1 = Arrays.stream(queue1).mapToLong(x -> x).sum();
        long t2 = Arrays.stream(queue2).mapToLong(x -> x).sum();
        long total = t1 + t2;
        
        if (total % 2 != 0) {
            return -1;
        }
        
        long target = total / 2;
        
        
        int[] q = new int[queue1.length + queue2.length];
        
        for (int i = 0; i < queue1.length; i++) {
            q[i] = queue1[i];
            q[i + queue1.length] = queue2[i];
        }
        
        int start = 0;
        int end = queue1.length - 1;
        int result = 0;
        
        while (t1 != target) {
            if (t1 > target) {
                t1 -= q[start];
                start++;
            } else {
                end += 1;
                if (end == q.length) {
                    return -1;
                }
                t1 += q[end];
            }
            result++;
        }
        return result;
    }
}